File: app/src/main/java/com/example/acer/movies/adapters/RecyclerViewAdapterTVShowCast.java
package com.example.acer.movies.adapters;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.acer.movies.models.Cast;
import com.example.acer.movies.R;
import com.example.acer.movies.network.URLConstants;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;

/**
 * Created by KeshavAggarwal on 13/03/17.
 */

public class RecyclerViewAdapterTVShowCast extends RecyclerView.Adapter<RecyclerViewAdapterTVShowCast.ViewHolder> {
    private ArrayList<Cast> mTvShowCast;
    Context mContext;

    public RecyclerViewAdapterTVShowCast(ArrayList<Cast> tvShowCastMain, Context context) {
        mTvShowCast = tvShowCastMain;
        mContext = context;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.tvshow_cast_list_item_view, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        if (mTvShowCast != null) {
            holder.tvshowCastName.setText(mTvShowCast.get(position).getName());
            Picasso.with(mContext).load(URLConstants.IMAGE_BASE_URL + mTvShowCast.get(position).getProfile_path()).into(holder.tvshowCastImage);
            String character = "as " + mTvShowCast.get(position).getCharacter();
            holder.tvshowCastCharacter.setText(character);
        }
    }

    @Override
    public int getItemCount() {
        return mTvShowCast.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView tvshowCastImage;
        TextView tvshowCastName;
        TextView tvshowCastCharacter;
        View tvshowCastDivider;

        ViewHolder(View itemView) {
            super(itemView);
            tvshowCastImage = (ImageView) itemView.findViewById(R.id.tvshowCastImage);
            tvshowCastName = (TextView) itemView.findViewById(R.id.tvshowCastName);
            tvshowCastCharacter = (TextView) itemView.findViewById(R.id.tvshowCastCharacter);
            tvshowCastDivider = (View) itemView.findViewById(R.id.tvshowCastDivider);

        }
    }
}

File: app/src/main/java/com/app/techvalley/movies/adapters/RecyclerViewAdapterTVShowCast.kt
package com.app.techvalley.movies.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.app.techvalley.movies.R
import com.app.techvalley.movies.models.Cast
import com.app.techvalley.movies.network.URLConstants
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.tvshow_cast_list_item_view.view.*
import java.util.*

/**
 * Created by KeshavAggarwal on 13/03/17.
 */

class RecyclerViewAdapterTVShowCast(private val mTvShowCast: ArrayList<Cast>?, internal var mContext: Context) : RecyclerView.Adapter<RecyclerViewAdapterTVShowCast.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.tvshow_cast_list_item_view, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        if (mTvShowCast != null) {
            holder.itemView.castNameTV.text = mTvShowCast[position].name
            Picasso.get().load(URLConstants.IMAGE_BASE_URL + mTvShowCast[position].profile_path).into(holder.itemView.castIV)
            val character = "as " + mTvShowCast[position].character
            holder.itemView.castCharacterTV.text = character
        }
    }

    override fun getItemCount(): Int {
        return mTvShowCast?.size ?: 0
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}
